src/v/pandaproxy/schema_registry/test/test_json_schema.cc

```diff
@@ -77,24 +77,27 @@ SEASTAR_THREAD_TEST_CASE(test_make_invalid_json_schema) {
     for (const auto& data : error_test_cases) {
         store_fixture f;
         BOOST_TEST_CONTEXT(data) {
-            BOOST_REQUIRE_EXCEPTION(
-              pps::make_canonical_json_schema(
-                f.store,
-                {pps::subject{"test"}, {data.def, pps::schema_type::json}})
-                .get(),
-              pps::exception,
-              [&data](pps::exception const& e) {
-                  auto code_matches = e.code() == data.err.code();
-                  if (code_matches && e.message() != data.err.message()) {
-                      fmt::print(
-                        "[{}] does not match expected [{}]\n",
-                        e.message(),
-                        data.err.message());
-                  }
-                  return code_matches;
-              });
+            try {
+                pps::make_canonical_json_schema(
+                  f.store,
+                  {pps::subject{"test"}, {data.def, pps::schema_type::json}})
+                  .get();
+            } catch (pps::exception const& e) {
+                BOOST_CHECK_EQUAL(e.code(), data.err.code());
+                BOOST_WARN_MESSAGE(
```

Comment:
It is intentional for ease of development since error messages tend to change while works is in progress. I'll change it back to CHECK once this set of PRs is done